plugins/match.hpp

```diff
@@ -261,12 +266,19 @@ template <class DataFacadeT> class MapMatchingPlugin : public BasePlugin
         osrm::matching::CandidateLists candidates_lists;
         const auto &input_coords = route_parameters.coordinates;
         const auto &input_timestamps = route_parameters.timestamps;
+        const auto &input_bearings = route_parameters.bearings;
         if (input_timestamps.size() > 0 && input_coords.size() != input_timestamps.size())
         {
             json_result.values["status"] = "Number of timestamps does not match number of coordinates .";
             return 400;
         }
 
+        if (input_bearings.size() > 0 && input_coords.size() != input_bearings.size())
+        {
+            json_result.values["status"] = "Number of bearings does not match number of coordinates .";
+            return 400;
+        }
+
```

Comment:
I see how this checks at least that the sizes are the same. But what happens if I do not interleave coordinated with bearings, but instead do a request such as:

> coordinate, bearing, coordinate, coordinate, bearing, bearing

then this check won't catch the mistake and from the last two bearings, only the last one will be used and there exists a bearing that is default initialized but never assigned to?